tests/test_dtypes.py

```diff
@@ -157,3 +183,56 @@ def test__get_gdal_dtype__int64():
     else:
         with pytest.raises(TypeError, match="Unsupported data type"):
             _get_gdal_dtype("int64")
+
+
+@pytest.mark.parametrize("dtype,nodata", itertools.product(DTYPES, [1, 127]))
+def test_write_mem(dtype, nodata):
```

Comment:
@adamjstewart @snowman2 @vincentsarago I would prefer that we not add more I/O round trip tests to this module. I added a test for the complex type, but that was me working in a rush, not the way I want to see the tests here evolve. What would you think about these tests going into tests/test_read_dtype.py (with a subsequent rename to test_rw_dtype.py)?

Reply:
So do you want me to rename this file or keep the file as is and move all of my changes to this new file?

Reply:
@sgillies can you clarify what you want me to do here?

Reply:
@adamjstewart please add the I/O tests to tests/test_read_dtype.py 🙏 Sorry about being vague!